crates/goose/src/agents/agent.rs

```diff
@@ -609,9 +609,16 @@ impl Agent {
     pub async fn create_recipe(&self, mut messages: Vec<Message>) -> Result<Recipe> {
         let extension_manager = self.extension_manager.lock().await;
         let extensions_info = extension_manager.get_extensions_info().await;
-        let system_prompt = self
-            .prompt_manager
-            .build_system_prompt(extensions_info, self.frontend_instructions.clone());
+
+        // Get model name from provider
+        let model_config = self.provider.get_model_config();
```

Comment:
For databricks wouldn't this just come from 

https://github.com/block/goose/blob/970147f8adfe505d08a5c3c6f5c1acd2ee88ec07/crates/goose/src/providers/databricks.rs#L239-L241

?

If the model is aliased, would we still get the required `gpt_4_1`?

Reply:
@alexhancock the modelconfig I think would be what user has configured, which would be the alias if I understand correctly your question (so yeah, still need to get it back, infact it was databricks that started me requiring this)

Reply:
and yeah - I was testing it with our "goose" databricks hosting endpoint, when it would return the appropriate model, and I checked it picked up the right promt.